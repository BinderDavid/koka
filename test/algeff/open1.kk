// test lifting of cps functions
function myid() : total (a -> <cps|e> a) {
  return fun(x){ x }
}

val hs : forall<e> list<int -> <cps|e> int> = [myid()]

function bar() {
  head(hs)(10)
}

function mymap( f, xs ) {
  match(xs) {
    Nil -> Nil
    Cons(x,xx) -> Cons(f(x),mymap(f,xx))
  }
}