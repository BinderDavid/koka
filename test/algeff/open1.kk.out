10
bar : () -> exn int
hs : forall<e> list<(int) -> e int>
main : () -> <exn,console> ()
myid : forall<a,e> () -> ((x : a) -> e a)